apps/pie-docs/src/_11ty/shortcodes/card.js

```diff
@@ -16,30 +20,101 @@ module.exports = function ({ items }) {
         },
     });
 
+    return isInternalLink ? internalLinkIcon : externalLinkIcon;
+};
+
+const buildCardLabel = (linkText, cardHasImage, href, shouldOpenInNewTab, isInternalLink) => {
+    const labelClasses = [
+        'c-card-labelContainer',
+        cardHasImage && 'c-card-labelContainer--hasImage',
+        isInternalLink && 'c-card-labelContainer--internalLink'
+    ].filter(Boolean).join(' ');
+
+    const target = shouldOpenInNewTab ? 'target="_blank"' : '';
+
+    return `<div class="${labelClasses}">
+                ${href
+        ? `<a class="c-card-label" href="${href}" ${target}><span>${linkText}</span></a>
```

Comment:
Feels like a little bit of duplication here with the classes and the span, and the ternary isn't super readable IMO (probably made worse by the line overflow)